operation/src/main/scala/io/vamp/operation/gateway/GatewayActor.scala

```diff
@@ -102,7 +101,7 @@ class GatewayActor extends ArtifactPaginationSupport with CommonSupportForActors
   private def checkRouteChanges(gateway: Gateway): Unit = {
     checked[Option[_]](actorFor[PersistenceActor] ? PersistenceActor.Read(gateway.name, classOf[Gateway])) map {
       case Some(old: Gateway) ⇒
-        compareNewRoutesAndGenerateEvents(old, gateway.routes, "routeChanged")
+        compareNewRoutesAndGenerateEvents(old, gateway, "routeChanged")
```

Comment:
This may also need checking validateOnly. Can you also test this?
You should test it in an internal gateway.

Reply:
Yeah, it's already checked in line 75. I tested it.